snoop/data/digests.py

```diff
@@ -90,46 +90,88 @@ def parent_id(file):
     return None
 
 
+def email_meta(digest_data):
+    def iter_parts(email_data):
+        yield email_data
+        for part in email_data.get('parts') or []:
+            yield from iter_parts(part)
+
+    email_data = digest_data['email']
+    headers = email_data['headers']
+
+    text_bits = []
+    for part in iter_parts(email_data):
+        part_text = part.get('text')
+        if part_text:
+            text_bits.append(part_text)
+
+    headers_to = []
+    for header in ['To', 'Cc', 'Bcc', 'Resent-To', 'Recent-Cc', 'Reply-To']:
+        headers_to += headers.get(header, [])
+
+    return {
+        'from': headers.get('From', [''])[0],
+        'to': headers_to,
+        'subject': headers.get('Subject', [''])[0],
+        'text': '\n\n'.join(text_bits).strip(),
+    }
+
+
 def get_document_data(digest):
     with digest.result.open() as f:
         digest_data = json.loads(f.read().decode('utf8'))
 
-    first_file = digest.blob.file_set.order_by('pk').first()
+    blob = digest.blob
+    first_file = blob.file_set.order_by('pk').first()
+    content = {
+        'content-type': blob.mime_type,
+        'filetype': filetype(blob.mime_type),
+        'text': digest_data.get('text'),
+        'md5': blob.md5,
+        'sha1': blob.sha1,
+        'size': blob.path().stat().st_size,
+        'filename': first_file.name,
+        'path': full_path(first_file),
+    }
 
-    return {
-        'id': digest.blob.pk,
+    rv = {
+        'id': blob.pk,
         'parent_id': parent_id(first_file),
         'version': zulu(digest.date_modified),
-        'content': {
-            'content-type': digest.blob.mime_type,
-            'filetype': filetype(digest.blob.mime_type),
-            'text': digest_data.get('text'),
-            'md5': digest.blob.md5,
-            'sha1': digest.blob.sha1,
-            'size': digest.blob.path().stat().st_size,
-            'filename': first_file.name,
-            'path': full_path(first_file),
-            '_emailheaders': digest_data.get('_emailheaders'),
-        },
+        'content': content,
     }
 
+    if blob.mime_type == 'message/rfc822':
+        rv['content'].update(email_meta(digest_data))
+
+        attachments = first_file.child_directory_set.first()
+        if attachments:
+            rv['children'] = [
+                child_file(f)
+                for f in attachments.child_file_set.order_by('pk').all()
+            ]
+
+    return rv
+
+
+def child_file(file):
```

Comment:
There's nothing specific to a `child` file here: maybe name it `file_to_dict` or something?